### ISSUE TITLE
c7n-orgs looking for subscriptions even though accounts exist

### ISSUE THREAD

**Issue by thomasattree** [capitalone/cloud-custodian#2436]
c7n-orgs cli is trying to yield azure subscriptions although AWS accounts should have already been yielded. I doubt you would want to yield both from the same config file and the schema also dictates oneOf accounts or subscriptions.

https://github.com/capitalone/cloud-custodian/commit/7e1bf77bdb6e87c3c4694ae37c011adadb47d416#diff-d0ad1d152e091597783f18cea409d819

accounts_iterator() should have more sanity to check for either accounts or subscriptions in  `tools/c7n_org/c7n_org/cli.py`

This is the traceback caused:
```
Traceback (most recent call last):
File "/root/virtualenvs/custodian/bin/c7n-org", line 11, in <module>
sys.exit(cli())
File "/root/virtualenvs/custodian/local/lib/python2.7/site-packages/click/core.py", line 722, in __call__
return self.main(*args, **kwargs)
File "/root/virtualenvs/custodian/local/lib/python2.7/site-packages/click/core.py", line 697, in main
rv = self.invoke(ctx)
File "/root/virtualenvs/custodian/local/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
return _process_result(sub_ctx.command.invoke(sub_ctx))
File "/root/virtualenvs/custodian/local/lib/python2.7/site-packages/click/core.py", line 895, in invoke
return ctx.invoke(self.callback, **ctx.params)
File "/root/virtualenvs/custodian/local/lib/python2.7/site-packages/click/core.py", line 535, in invoke
return callback(*args, **kwargs)
File "/root/virtualenvs/custodian/local/lib/python2.7/site-packages/c7n_org/cli.py", line 501, in run
for a in accounts_iterator(accounts_config):
File "/root/virtualenvs/custodian/local/lib/python2.7/site-packages/c7n_org/cli.py", line 409, in accounts_iterator
for a in config.get('subscriptions'):
```

**Comment by thisisshi** [comment]
What's the error on the traceback? Looks like you're missing the last line there

**Comment by thisisshi** [comment]
also, the oneof schema prevents the policy form running at all as it's validated prior to execution

**Comment by kapilt** [comment]
added a slightly simpler fix along with better gcp support in #2438
